pkg/apis/v1beta1/cluster.go

```diff
@@ -175,20 +185,27 @@ func DefaultAPISpec() *APISpec {
 	// Collect all nodes addresses for sans
 	addresses, _ := util.AllAddresses()
 	publicAddress, _ := util.FirstPublicAddress()
+	addresses = append(addresses, publicAddress)
+	addresses = util.Unique(addresses)
```

Comment:
This is the fix for #520, right? Might be nice to have in a separate commit since it does not have anything to do with empty values.